app/src/main/java/com/epfl/Utils/drawyourpath/Utils.kt

```diff
@@ -33,23 +38,128 @@ object Utils {
     }
 
     /**
-     * Decodes the photo from base64 string to bitmap format and return null if the dataSnapShot is null
+     * get the profile photo as bitmap or if none the default profile photo
+     * @param photo the photo to decode
+     * @param res the resources used to get the default profile photo
+     * @return the profile photo or the default profile photo
+     */
+    fun decodePhotoOrGetDefault(photo: ByteArray?, res: Resources): Bitmap {
+        return photo?.let { decodePhoto(photo) } ?: getDefaultPhoto(res)
+    }
+
+    /**
+     * Decodes the photo from base64 string to bitmap format
      * @param photoStr photo encoded
-     * @return the photo in bitmap format, and null if no photo is stored on the database
+     * @return the photo in bitmap format
      */
-    fun decodePhoto(photoStr: String): Bitmap {
-        val tabByte = Base64.getDecoder().decode(photoStr as String)
-        return BitmapFactory.decodeByteArray(tabByte, 0, tabByte.size)
+    fun decodePhotoOrGetDefault(photoStr: String?, res: Resources): Bitmap {
+        return decodePhotoOrGetDefault(decodeStringAsByteArray(photoStr), res)
     }
 
     /**
-     * Encodes a bitmap image.
-     * @return A base64 string of the image.
+     * Decodes the photo from byte array to bitmap format
+     * @param photo photo encoded
+     * @return the photo in bitmap format
      */
-    fun encodePhoto(photo: Bitmap, quality: Int = 50): String {
+    fun decodePhoto(photo: ByteArray): Bitmap? {
+        return BitmapFactory.decodeByteArray(photo, 0, photo.size)

```

Comment:
While testing, this function seems to throw sometimes when the conversion fails. Not sure if we want to handle that here. I let you decide.

Reply:
This function can throw IndexOutOfBound if the size is bigger than the bytearray (which should never happen here) or IllegalArgumentException but only if some options are specified, here they are not specified so should also never happen.